### DISCUSSION TITLE
Debugger.Launch() in source generator code

### DISCUSSION THREAD

**Discussion by zbesevic** [dotnet/roslyn#50606] [Q&A]
Hi. I'm writing a code generator which I try debugging with Debugger.Launch(). Before the compiler builds my project (I'm building the whole solution) which needs code generation ("references" the source generator project)I m getting devenv.exe unhandled exception popping up and VS tries to connect to the debugger. When I attach, the debugger goes through my generator but it should only for my project which needs generation. When the compiler compiles the project which needs code generation then the process which pops up to attach to is csc.exe which is the compiler. What are these attempts to debug unhandled exceptions from devenv.exe? Also, VBCSCompiler.exe throws this unhandled exception. It asks for admin rights to run the debugger.

**Comment by michael-freidgeim-webjet** [comment]
One of described workarounds https://turnerj.com/blog/the-pain-points-of-csharp-source-generators (not very convenient)
